### PR TITLE
Optimize build from source

### PR DESCRIPTION
**Pull request by gasinvein**
- Remove `npm_config_nodedir` since we need native modules for electron
- Don't disable rebuilding native modules
- Remove redundant environment variables
- Use `jq` to edit package.json
- Use `.yarnrc` files instead of passing `--offline` cli arg
- Add empty lines between modules and sources for readability

### CODE REVIEW

File: im.riot.Riot.yaml [hunk 1]
```diff
@@ -158,45 +153,40 @@ modules:
         commit: 245d13875faaecb7c3fec2df112bf3a7a6e8fec9
         dest: main
         x-checker-data:
-          type: git
-          tag-pattern: ^v([\\d.]+)$
-          version-scheme: semantic
+          type: json
```
Comment (jwillikers) [inline im.riot.Riot.yaml:156]: I'm just curious about why you choose the `json` `x-checker-data`?

File: im.riot.Riot.yaml [hunk 2]
```diff
@@ -158,45 +153,40 @@ modules:
...
-      - sed -i 's/yarn/yarn --offline/g' package.json
-
       # Patch the electronVersion to build with to match the minimum version in the dependency list.
-      - |
-        sed -i 's/"electronVersion": "13.1.6"/"electronVersion": "13.1.7"/' package.json
-
-      # Rebuild native modules by electron-rebuild
-      # https://github.com/electron-userland/electron-builder/issues/4100
-      - sed -i 's/"build":\ {/"build":\ {\n"npmRebuild":\ false,/' package.json
+      - jq 'del(.build.electronVersion)' <<<$(cat package.json) > package.json
```
Comment (jwillikers) [inline im.riot.Riot.yaml:180]: Using `jq` is much better idea...

**Review by jwillikers — APPROVED** [review]
Thanks for all the work, things look *way better* considering it was a bit of a mess.

File: im.riot.Riot.yaml (hunk 1 above)
Comment (gasinvein) [reply im.riot.Riot.yaml:156]: Because json checker can get stable timestamps, unlike git checker, see flathub/flatpak-external-data-checker#154

**Pull request merged**